src/frontend/js/pages/DashboardPageNotFound/index.tsx

```diff
@@ -0,0 +1,56 @@
+import { FormattedMessage, defineMessages } from 'react-intl';
+import { Link, Navigate, useLocation, useSearchParams } from 'react-router-dom';
+import { LearnerDashboardPaths } from 'widgets/Dashboard/utils/learnerRoutesPaths';
+import { TeacherDashboardPaths } from 'widgets/Dashboard/utils/teacherDashboardPaths';
+
+const messages = defineMessages({
+  pageNotFoundTitle: {
+    id: 'page.PageNotFound.pageNotFoundTitle',
+    description: 'Title of the page not found page',
+    defaultMessage: '404, Page not found!',
+  },
+  pageNotFoundSubTitle: {
+    id: 'page.PageNotFound.pageNotFoundSubTitle',
+    description: 'Subtitle of the page not found page',
+    defaultMessage: "The page you're looking for doesn't exist.",
+  },
+  pageNotFoundHomeLink: {
+    id: 'page.PageNotFound.pageNotFoundHomeLink',
+    description: 'Home link of the page not found page',
+    defaultMessage: 'Back to the home page.',
+  },
+});
+
+export const NavigatePageNotFound = () => {
+  const location = useLocation();
+  return <Navigate to={`/page-not-found?from=${encodeURI(location.pathname)}`} />;
+};
+
+const DashboardPageNotFound = () => {
+  const [urlSearchParams] = useSearchParams();
+  let from = urlSearchParams.get('from');
+  from = from ? decodeURI(from) : '';
+  const redirectPath = from.includes('/teacher/')
+    ? TeacherDashboardPaths.ROOT
+    : LearnerDashboardPaths.COURSES;
+
+  return (
+    <div className="page-not-found">
+      <div className="page-not-found__wrapper">
+        <h1 className="page-not-found__title">
+          <FormattedMessage {...messages.pageNotFoundTitle} />
+        </h1>
+        <p className="page-not-found__subtitle">
+          <FormattedMessage {...messages.pageNotFoundSubTitle} />
+        </p>
+        <p className="page-not-found__link">
+          <Link to={redirectPath}>
+            <FormattedMessage {...messages.pageNotFoundHomeLink} />
+          </Link>
+        </p>
+      </div>
+    </div>
+  );
+};
```

Comment:
IMO this is overengineer. I'll just put a link to the home page like the link label refers.